pkg/providers/vsphere/vmprovider_vm_test.go

```diff
@@ -3209,6 +3207,90 @@ func vmTests() {
 						Expect(currentSnap).ToNot(BeNil())
 						Expect(currentSnap.Name).To(Equal(vmSnapshot.Name))
 					})
+
+					Context("and the snapshot was taken when VM was powered on and is now powered off", func() {
+						It("should successfully power on the VM after reverting to a Snapshot in PoweredOn state", func() {
```

Comment:
Hmnm.  I don't quite grok this.
You are creating two snapshots, both _without_ memory, so the power state of the snapshot should be powered off.  And therefore, when the VM is reverted, it should be in powered off state?